main.go

```diff
@@ -95,3 +109,35 @@ func main() {
 		klog.Fatalf("unable to run the manager: %v", err)
 	}
 }
+
+// createClientConfigs allow users to provide second config using management-kubeconfig, if specified
+// try to build it from provided path. First returned value is management config used for Machines,
+// second is workload config used for Node/CSRs.
+func createClientConfigs(managementClientPath string) (*rest.Config, *rest.Config, error) {
+	if managementClientPath == "" {
+		return control.GetConfigOrDie(), control.GetConfigOrDie(), nil
+	}
+
+	managementConfig, err := clientcmd.BuildConfigFromFlags("", managementClientPath)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return managementConfig, control.GetConfigOrDie(), nil
```

Comment:
would this get the workload kubeconfig in-cluster? This binary is expected to run in the management cluster so the workload config would come from a secret and need to be used to manipulate nodes and watch CSR.

Reply:
IIUC we already have a `kubeconfig` flag which allows you to override the workload config, so I think that answers running in the management cluster with a secret pointing to the workload cluster, but then it seems we have to also specify the management cluster as a kubeconfig as well? 

I think we need to have some way for the management config to be used as in cluster creds, while the workload is using a secret based kubeconfig. I think in the autoscaler there's a special flag for this, we should copy their pattern

Reply:
I tried to follow their pattern https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/clusterapi/clusterapi_provider.go#L145

In the case of default openshift deployment, nothing changes because management config == workload config. For the case when we need separate clients: management cluster config is obtained using `--management-kubeconfig` flag and workload using `--kubeconfig`.